src/screens/DiscoveryScreenV2/fragment/DomainFragment.js

```diff
@@ -1,182 +1,279 @@
-            const newFollowedDomains = [...followedDomains]
-            newFollowedDomains[index].user_id_follower = willFollow ? myId : null
+  const domains = React.useMemo(() => {
+    return discovery.initialDomains.map((item) => ({
+      ...item,
+      following: item.following !== undefined ? item.following : item.user_id_follower !== null
+    }));
+  }, [discovery.initialDomains]);
+
+  React.useEffect(() => {
+    const parseToken = async () => {
+      const id = await getUserId();
+      if (id) {
+        setMyId(id);
+      }
+    };
+    parseToken();
+  }, []);
 
-            // DiscoveryAction.setNewFollowedDomains(newFollowedDomains, discoveryDispatch)
-            setFollowedDomains(newFollowedDomains)
+  const __handleOnPressDomain = (item) => {
+    const navigationParam = {
+      item: {
+        content: {
+          domain_page_id: item.domain_page_id
+        },
+        domain: {
+          image: item.logo,
+          credderScore: item?.credder_score
+        },
+        og: {
+          domain: item.domain_name,
+          domainImage: item.logo
         }
+      }
+    };
 
-        if (from === FROM_UNFOLLOWED_DOMAIN) {
-            const newUnfollowedDomains = [...unfollowedDomains]
-            newUnfollowedDomains[index].user_id_follower = willFollow ? myId : null
+    navigation.push('DomainScreen', navigationParam);
+  };
 
-            // DiscoveryAction.setNewUnfollowedDomains(newUnfollowedDomains, discoveryDispatch)
-            setUnfollowedDomains(newUnfollowedDomains)
-        }
+  const __handleFollow = async (from, willFollow, item, index) => {
+    if (from === FROM_FOLLOWED_DOMAIN_INITIAL) {
+      console.log('from 0');
+      const newFollowedDomains = [...domains];
+      newFollowedDomains[index].following = !!willFollow;
+      newFollowedDomains[index].user_id_follower = myId;
 
-        const data = {
-            domainId: item.domain_page_id,
-            source: 'discoveryScreen',
-        };
+      console.log('newFollowedDomains', newFollowedDomains[index]);
 
-        if (willFollow) {
-            await followDomain(data);
-        } else {
-            await unfollowDomain(data);
-        }
+      FollowingAction.setFollowingDomain(newFollowedDomains, followingDispatch);
+      DiscoveryAction.setDiscoveryInitialDomains(newFollowedDomains, discoveryDispatch);
+    }
+    if (from === FROM_FOLLOWED_DOMAIN) {
+      console.log('from 1');
+      const newFollowedDomains = [...followedDomains];
+      newFollowedDomains[index].following = !!willFollow;
+      newFollowedDomains[index].user_id_follower = myId;
+
+      setFollowedDomains(newFollowedDomains);
+    }
+
+    if (from === FROM_UNFOLLOWED_DOMAIN) {
+      console.log('from 2');
+      const newUnfollowedDomains = [...unfollowedDomains];
+      newUnfollowedDomains[index].following = !!willFollow;
+      newUnfollowedDomains[index].user_id_follower = myId;
+
+      setUnfollowedDomains(newUnfollowedDomains);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There are repeated code blocks for updating the state of followed and unfollowed domains. This violates the DRY (Don't Repeat Yourself) principle. Consider creating a helper function to update the state based on the parameters.

```diff
+   const updateDomains = (from, willFollow, item, index, domains, setDomains) => {
+     const newDomains = [...domains];
+     newDomains[index].following = !!willFollow;
+     newDomains[index].user_id_follower = myId;
+
+     if (from === FROM_FOLLOWED_DOMAIN_INITIAL) {
+       FollowingAction.setFollowingDomain(newDomains, followingDispatch);
+       DiscoveryAction.setDiscoveryInitialDomains(newDomains, discoveryDispatch);
+     } else {
+       setDomains(newDomains);
+     }
+   };

  const __handleFollow = async (from, willFollow, item, index) => {
    if (from === FROM_FOLLOWED_DOMAIN_INITIAL) {
      updateDomains(from, willFollow, item, index, domains, null);
    }
    if (from === FROM_FOLLOWED_DOMAIN) {
      updateDomains(from, willFollow, item, index, followedDomains, setFollowedDomains);
    }
    if (from === FROM_UNFOLLOWED_DOMAIN) {
      updateDomains(from, willFollow, item, index, unfollowedDomains, setUnfollowedDomains);
    }
    // rest of the code...
  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->